x-pack/test/alerting_api_integration/spaces_only/tests/alerting/find.ts

```diff
@@ -171,6 +189,62 @@ const findTestUtils = (
         expect(response.body.total).to.equal(1);
         expect(response.body.data[0].params.strValue).to.eql('my b');
       });
+
+      it('should sort by parameters', async () => {
+        const response = await supertest.get(
+          `${getUrlPrefix(Spaces.space1.id)}/${
+            describeType === 'public' ? 'api' : 'internal'
+          }/alerting/rules/_find?sort_field=params.severity&sort_order=asc`
+        );
+        expect(response.body.data[0].params.severity).to.equal('low');
+        expect(response.body.data[1].params.severity).to.equal('medium');
+        expect(response.body.data[2].params.severity).to.equal('high');
+      });
+
+      it('should search by parameters', async () => {
+        const response = await supertest.get(
+          `${getUrlPrefix(Spaces.space1.id)}/${
+            describeType === 'public' ? 'api' : 'internal'
+          }/alerting/rules/_find?search_fields=params.severity&search=40-medium`
+        );
+
+        expect(response.status).to.eql(200);
+        expect(response.body.total).to.equal(1);
+        expect(response.body.data[0].params.severity).to.eql('medium');
+      });
+
+      it('should filter on parameters', async () => {
+        const response = await supertest.get(
+          `${getUrlPrefix(Spaces.space1.id)}/${
+            describeType === 'public' ? 'api' : 'internal'
+          }/alerting/rules/_find?filter=alert.attributes.params.risk_score:40`
+        );
+
+        expect(response.status).to.eql(200);
+        expect(response.body.total).to.equal(1);
+        expect(response.body.data[0].params.risk_score).to.eql(40);
+
+        if (describeType === 'public') {
```

Comment:
😍

Reply:
Looks great in practice! (The separate of public and internal)